blockchain/state_manager/src/lib.rs

```diff
@@ -1115,61 +1117,59 @@ where
         let sm_cloned = Arc::clone(self);
 
         // Wait for message to be included in head change.
-        let mut subscriber_poll = task::spawn::<
-            _,
-            Result<(Option<Arc<Tipset>>, Option<MessageReceipt>), Error>,
-        >(async move {
-            loop {
-                match subscriber.recv().await {
-                    Ok(subscriber) => match subscriber {
-                        HeadChange::Revert(_tipset) => {
-                            if candidate_tipset.is_some() {
-                                candidate_tipset = None;
-                                candidate_receipt = None;
-                            }
-                        }
-                        HeadChange::Apply(tipset) => {
-                            if candidate_tipset
-                                .as_ref()
-                                .map(|s| tipset.epoch() >= s.epoch() + confidence)
-                                .unwrap_or_default()
-                            {
-                                return Ok((candidate_tipset, candidate_receipt));
-                            }
-                            let poll_receiver = receiver.try_recv();
-                            if let Ok(Some(_)) = poll_receiver {
-                                block_revert
-                                    .write()
-                                    .await
-                                    .insert(tipset.key().to_owned(), true);
+        let mut subscriber_poll =
+            task::spawn::<_, Result<(Option<Arc<Tipset>>, Option<Receipt>), Error>>(async move {
```

Comment:
While we're at this, I think we can get rid of the type annotation here

Reply:
done!